3rdParty/fuerte/src/HttpConnection.cpp

```diff
@@ -205,7 +204,12 @@ void HttpConnection<ST>::shutdownConnection(const ErrorCondition ec) {
   FUERTE_LOG_CALLBACKS << "shutdownConnection\n";
   
   _state.store(State::Disconnected, std::memory_order_release);
-  _timeout.cancel();    // cancel timeouts
+  try {
```

Comment:
can you just pass in a reference to `asio_ns::error_code`

Reply:
instead of try..catch

Reply:
I don't think so.
According to the [boost docs](https://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/basic_waitable_timer/cancel/overload2.html) calling cancel with an error code is deprecated and shouldn't be used.
And it will not help if `cancel` throws and is called from HttpConnection's destructor. We must prevent exceptions from propagating from out of the destructor, otherwise the program will just abort.